src/app/Sidebar/SidebarUsers.js

```diff
@@ -3,30 +3,43 @@ import React, { Component, PropTypes } from 'react';
 import filter from 'lodash/filter';
 import find from 'lodash/find';
 import groupBy from 'lodash/groupBy';
+import startsWith from 'lodash/startsWith';
 import map from 'lodash/map';
 import size from 'lodash/size';
 import { Link } from 'react-router';
-import without from 'lodash/without';
 
-import './SidebarMessages.scss';
+import Icon from '../../widgets/Icon';
+
+import './SidebarUsers.scss';
 
 function UnreadCount({ unread }) {
   if (!unread) return null;
   return (
-    <span className="SidebarMessages__unreadCount">
+    <span className="SidebarUsers__unreadCount">
       <span>
         {unread}
       </span>
     </span>
   );
 }
 
-export default class SidebarMessages extends Component {
+export default class SidebarUsers extends Component {
   static propTypes = {
     contacts: PropTypes.array,
     messages: PropTypes.object,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      search: '',
+    };
+  }
+
+  search = (e) => {
+    this.setState({ search: e.target.value });
+  };
+
   render() {
```

Comment:
So many calculations are happening in the render() function. It's better to isolate calculations into different pure functions (could be defined inside the scope of react class or in the top of file but better not in a separate file) with meaningful names so with a quick look into render functions we understand the stream of actions taking place.

Functions could be defined with verbose names like `usersFilteredBySearchQuery()` or in the similar ways to make it clean and understandable.